mmpose/datasets/datasets/body3d/body3d_h36m_dataset.py

```diff
@@ -0,0 +1,253 @@
+        8: 'thorax',
+        9: 'neck_base',
+        10: 'head',
+        11: 'left_shoulder',
+        12: 'left_elbow',
+        13: 'left_wrist',
+        14: 'right_shoulder',
+        15: 'right_elbow',
+        16: 'right_wrist'
+
+    Args:
+        PoseTransferBaseDataset ([type]): [description]
+    """
+
+    JOINT_NAMES = [
+        'Root', 'LHip', 'LKnee', 'LFoot', 'RHip', 'RKnee', 'RFoot', 'Spine',
+        'Thorax', 'NeckBase', 'Head', 'LShoulder', 'LElbow', 'LWrist',
+        'RShoulder', 'RElbow', 'RWrist'
+    ]
+
+    JOINT_IDX_GLOBAL = [
+        14, 2, 1, 0, 3, 4, 5, 16, 12, 17, 18, 9, 10, 11, 8, 7, 6
+    ]
+    JOINT_NUM_GLOBAL = 24
+
+    # 2D joint source options:
+    # "gt": from the annotation file
+    # "detection": from a detection result file of 2D keypoint
+    # "pipeline": will be generate in the pipeline
+    SUPPORTED_JOINT_2D_SRC = {'gt', 'detection', 'pipeline'}
+
+    # metric
+    ALLOWED_METRICS = {'joint_error'}
+
+    def load_config(self, data_cfg):
+        super().load_config(data_cfg)
+        # h36m specific attributes
+        self.joint_2d_src = data_cfg.get('joint_2d_src', 'gt')
+        if self.joint_2d_src not in self.SUPPORTED_JOINT_2D_SRC:
+            raise ValueError(
+                f'Unsupported joint_2d_src "{self.joint_2d_src}". '
+                f'Supported options are {self.SUPPORTED_JOINT_2D_SRC}')
+
+        self.joint_2d_det_file = data_cfg.get('joint_2d_det_file', None)
+
+        self.need_camera_param = data_cfg.get('need_camera_param', False)
+        if self.need_camera_param:
+            assert 'camera_param_file' in data_cfg
+            self.camera_param = self._load_camera_param(
+                data_cfg['camera_param_file'])
+
+        # h36m specific annotation info
+        ann_info = {}
+        ann_info['flip_pairs'] = [[1, 4], [2, 5], [3, 6], [11, 14], [12, 15],
+                                  [13, 16]]
+        ann_info['upper_body_ids'] = (0, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16)
+        ann_info['lower_body_ids'] = (1, 2, 3, 4, 5, 6)
+        ann_info['use_different_joint_weights'] = False
+
+        self.ann_info.update(ann_info)
+
+    def load_annotation(self):
+        data_info = super().load_annotation()
+
+        # convert 3D joints from original 24-keypoint to standard 17-keypoint
+        assert data_info['joint_3d'].shape[1] == self.JOINT_NUM_GLOBAL
+        data_info['joint_3d'] = data_info['joint_3d'][:, self.JOINT_IDX_GLOBAL]
+
+        # get 2D joints
+        if self.joint_2d_src == 'gt':
+            assert data_info['joint_2d'].shape[1] == self.JOINT_NUM_GLOBAL
+            data_info['joint_2d'] = data_info['joint_2d'][:, self.
+                                                          JOINT_IDX_GLOBAL]
+        elif self.joint_2d_src == 'detection':
+            data_info['joint_2d'] = self._load_joint_2d_detection(
+                self.joint_2d_det_file)
+        elif self.joint_2d_src == 'pipeline':
+            pass
```

Comment:
Add an assertion on the presence of 'joint_2d'

Reply:
Oh I misunderstood the code. Simply ignore it :)

There is no need to add assertion in L110